---
title: "Kevin BEL-A RNA-seq"
output:
  pdf_document: default
  html_document: default
date: "2024-07-11"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(DESeq2)
library(ggplot2)
library(ggrepel)
library(tidyverse)
library(EnsDb.Hsapiens.v86)
library(tximport)
library(dplyr)
library(EnhancedVolcano)
```

```{r}
###### Creating tx2gene from human Ensembl build ##########
edb <- EnsDb.Hsapiens.v86
tx2gene <- transcripts(edb, return.type = "DataFrame")
tx2gene <- tx2gene[,c(7,8)]
tx2gene <- tx2gene[,c(2,1)]
```


```{r}
######### Reading in files with tximport ########################
dir <- ("/Volumes/biekerlab_backup/Kevin/output")
list.files(dir)
samples <- read.table(file.path(dir, "samples.txt"), header = TRUE)
files <- file.path(dir, samples$Samples, "quant.sf")
files
names(files) <- c("CDA_1","CDA_2","CDA_3","WT_1","WT_2","WT_3")
all(file.exists(files))
txi <- tximport(files, type = "salmon", tx2gene = tx2gene, ignoreTxVersion = TRUE)
names(txi)
head(txi$counts)
```

```{r}
######### Creating the DESeq2 DF #########################
samples$condition <- factor(rep(c("CDA","WT"),each=3))
samples
ddsTxi <- DESeqDataSetFromTximport(txi,colData = samples,design = ~ condition)
```

```{r}
########## pre-filtering ###############
keep <- rowSums(counts(ddsTxi)) >= 10
dds <- ddsTxi[keep,]
```

```{r}
######## Differential expression #########
dds <- DESeq(dds)
res <- results(dds, contrast=c("condition","CDA","WT"))
head(res)
```

```{r}
##### Ordering and exporting results #########
resOrdered <- res[order(res$pvalue),]
summary(res)
resOrdered$Gene_id <- rownames(resOrdered)
resOrdered <- resOrdered[,c(7,1,2,3,4,5,6)]
write.csv(resOrdered, "deseq_results.csv")
```

```{r}
######## Get names ##########################
res_genes <- select(edb, keys = resOrdered$Gene_id, keytype = "GENEID", columns = c("GENEID", "GENENAME"))
res_final <- cbind(res_genes,resOrdered[,c(2:7)])
```

```{r}
######### Filtering #####################
#### using dplyr::filter so need to detach ensembldb ##################
up_sig_CDA <- dplyr::filter(as.data.frame(res_final), padj < 0.05 & log2FoldChange > 1)
write.csv(up_sig_CDA, "up_sig_CDA.csv")
down_sig_CDA <- dplyr::filter(as.data.frame(res_final), padj < 0.05 & log2FoldChange < -1)
write.csv(down_sig_CDA, "down_sig_CDA.csv")
```

```{r}
########## Variance stabilized Normalization ##################
vsd <- vst(dds, blind = FALSE)
var_stab_data <- assay(vsd)
var_stab_data <- as.data.frame(var_stab_data)
var_stab_data$Gene_id <- rownames(var_stab_data)
var_stab_data <- var_stab_data[,c(7,1,2,3,4,5,6)]
vsgenes <- select(edb, keys = var_stab_data$Gene_id, keytype = "GENEID", columns = c("GENEID", "GENENAME"))
var_stab_data <- cbind(vsgenes,var_stab_data[,c(2:7)])
```

```{r}
######### Generating normalized counts #######################
norm_count_salmon <- counts(dds, normalized = TRUE)
norm_count_salmon <- cbind(vsgenes,norm_count_salmon[,c(1:6)])
write.csv(as.data.frame(norm_count_salmon), file = "deseq_norm_counts.csv")
```

```{r}
######### Graphing PCA #######################
plotPCA(vsd, intgroup="condition") + theme_classic() + theme(aspect.ratio = 1)
ggsave("~/Desktop/pca.png")
```

```{r}
######### Graphing Volcano #######################
res_sig <- rbind(up_sig_CDA, down_sig_CDA)
View(res_sig)
write.csv(res_sig, "~/Desktop/res_sig.csv",quote=FALSE,row.names = FALSE)
EnhancedVolcano(res_sig, lab = res_sig$GENENAME, x = 'log2FoldChange', y = 'padj', cutoffLineWidth = 0, gridlines.major = FALSE, gridlines.minor = FALSE) + xlim(-10,15) + theme(legend.position="none")
ggsave("~/Desktop/volcano.png")
```
